extensions-core/azure-extensions/src/main/java/org/apache/druid/data/input/azure/AzureEntity.java

```diff
@@ -40,21 +42,32 @@ public class AzureEntity extends RetryingInputEntity
 {
   private final CloudObjectLocation location;
   private final AzureByteSource byteSource;
+  private final String scheme;
 
   @AssistedInject
   AzureEntity(
       @Nonnull @Assisted CloudObjectLocation location,
+      @Nonnull @Assisted AzureStorage azureStorage,
+      @Nonnull @Assisted String scheme,
       @Nonnull AzureByteSourceFactory byteSourceFactory
+
   )
   {
     this.location = location;
-    this.byteSource = byteSourceFactory.create(location.getBucket(), location.getPath());
+    this.scheme = scheme;
+    // If scheme is azureStorage, containerName is the first prefix in the path, otherwise containerName is the bucket
+    if (AzureStorageAccountInputSource.SCHEME.equals(this.scheme)) {
+      Pair<String, String> locationInfo = AzureStorageAccountInputSource.getContainerAndPathFromObjectLocation(location);
```

Comment:
Should the returned locationInfo be validated in anyway? What if the containerName or blobPath are missing?